### ISSUE TITLE
Add detailed instructions for developing SymPy with Conda

### ISSUE THREAD

**Issue by certik** [sympy/sympy#8790]
```
$ wget http://repo.continuum.io/miniconda/Miniconda-3.7.0-Linux-x86_64.sh
$ bash Miniconda-3.7.0-Linux-x86_64.sh # install into /local/certik/miniconda
$ cd ~/bin
$ ln -s /local/certik/miniconda/bin/conda .
$ ln -s /local/certik/miniconda/bin/activate .
$ ln -s /local/certik/miniconda/bin/deactivate .
```
Add `~/bin` into my $PATH in `.bashrc`.

Then let's create a Python 3.4 environment and activate it:
```
$ conda create -n py34 python=3.4 anaconda
$ source activate py34
```
Finally, let's add mpmath to it, so that we can use sympy:
```
$ conda install mpmath
Fetching package metadata: ..
Error: No packages found in current linux-64 channels matching: mpmath

Did you mean one of these?

    llvmmath

You can search for this package on Binstar with

    binstar search -t conda mpmath
```
So this currently fails. We can search on binstar, as it suggests:
```
$ binstar search -t conda mpmath
Run 'binstar show <USER/PACKAGE>' to get more details:
Packages:
                          Name | Access       | Package Types   | Summary             
     ------------------------- | ------------ | --------------- | --------------------
               asmeurer/mpmath | public       | conda           | Python library for arbitrary-precision floating-point arithmetic
                   auto/mpmath | published    | conda           | http://mpmath.googlecode.com
Found 2 packages
```
Then we do:
```
$ binstar show asmeurer/mpmath
Using binstar api site https://api.binstar.org
Name:    mpmath
Summary: Python library for arbitrary-precision floating-point arithmetic
Access:  public
Package Types:  conda
Versions:
   + 0.19

To install this package with conda run:
     conda install --channel https://conda.binstar.org/asmeurer mpmath
```
So finally, we can do:
```
$ conda install --channel https://conda.binstar.org/asmeurer mpmath
Fetching package metadata: ...
Error: No packages found in current linux-64 channels matching: mpmath

Did you mean one of these?

    llvmmath

You can search for this package on Binstar with

    binstar search -t conda mpmath
```
And it still fails, I assume because I use 64 bit linux? In any case, this needs to be fixed and step by step instructions added so that people can easily keep developing sympy.

**Comment by asmeurer** [comment]
Your ln steps are unusual. For most people, the correct step is to let the installer edit the bash profile to add Miniconda to the PATH.

**Comment by asmeurer** [comment]
OK, mpmath is now on the official Continuum repos. So you can just `conda install mpmath`.

**Comment by certik** [comment]
Excellent, I tested it, and it works. Here are the instructions:
```
$ conda create -n py34 python=3.4 mpmath ipython
$ source activate py34
```
And sympy works, from the git repository.

**Comment by skirpichev** [comment]
conda itself is a non-standard tool (e.g. in the Debian).

I think, we should promote also more usual variants with virtualenv/pip.

**Comment by asmeurer** [comment]
I personally care more about tools that work well and minimize the amount of work we have to do to help people set up their development environment than about being "standard" (whatever that means). Conda is better here than pip, especially when you start to add Windows to the equation (and very especially if someone needs to install a package that isn't pure Python like numpy or the IPython notebook).

**Comment by certik** [comment]
Exactly. Or CSymPy... Conda is a solution that fixes all these issues. Hashdist too. But pip doesn't.

**Comment by skirpichev** [comment]
On Mon, Jan 12, 2015 at 02:44:08PM -0800, Ondřej Čertík wrote:
>    But pip doesn't.

Hmm, pip works for me in the Debian out of the box, but
not Conda (which is external software).

**Comment by moorepants** [comment]
Yes, pip is fine for sympy + mpmath. But if you want to use some of SymPy modules that depend on Cython, Theano, IPython, etc conda makes it extremely easier to deal with those dependencies. You can potentially use the Debian package manager for those too, but you'll often be stuck with older versions that you may prefer.

We should have instructions for both pip/virtualenv and conda based development setup.

**Comment by skirpichev** [comment]
> pip is fine for sympy + mpmath

It works for me in case of any software from the scipy stack.

**Comment by moorepants** [comment]
> It works for me in case of any software from the scipy stack.

No one is claiming that pip can't work for the SciPy stack. In fact, I'm able to install pretty much all software in the SciPy stack via pip too, but it takes a great deal more effort to properly specify all of the dependencies. The other annoying issue, is that for every single virtualenv that you create you have to recompile all of the dependencies or try to hack it with symlinks (which doesn't always work, try building wxpython or mayavi in a virtualenv, for example). Conda is a fantastic solution, not only for these issues but for others too. Just because you don't have these issues, doesn't mean other people don't.

**Comment by oscarbenjamin** [comment]
Times have changed. Pip can now install most things without compiling anything.
